hudi-common/src/main/java/org/apache/hudi/common/table/timeline/MetadataConversionUtils.java

```diff
@@ -338,6 +338,15 @@ public static HoodieArchivedMetaEntry createMetaWrapperForEmptyInstant(HoodieIns
     return archivedMetaWrapper;
   }
 
+  private static <T extends HoodieCommitMetadata> Option<T> getArchivedCommitMetadata(HoodieTableMetaClient metaClient, HoodieInstant instant, Class<T> clazz) throws IOException {
+    T commitMetadata = metaClient.getArchivedTimeline().readInstantContent(instant, clazz);
```

Comment:
Why does this re-fetch the data instead of using the `instantDetails` that are already present above?

Reply:
reverted this change